plugins/inputs/dpdk/dpdk.go

```diff
@@ -0,0 +1,240 @@
+		dpdk.Log.Warn("DPDK plugin is enabled and configured with nothing to read")
+	}
+
+	dpdk.connector = newDpdkConnector(dpdk.SocketPath, dpdk.AccessTimeout)
+	result, err := dpdk.connector.connect()
+	if result != "" {
+		dpdk.Log.Info(result)
+	}
+	return err
+}
+
+// Checks that user-supplied commands are unique and match DPDK commands format
+func (dpdk *dpdk) validateCommands() error {
+	dpdk.AdditionalCommands = uniqueValues(dpdk.AdditionalCommands)
+
+	for _, commandWithParams := range dpdk.AdditionalCommands {
+		if len(commandWithParams) == 0 {
+			return fmt.Errorf("got empty command")
+		}
+
+		if commandWithParams[0] != '/' {
+			return fmt.Errorf("'%v' command should start with '/'", commandWithParams)
+		}
+
+		if commandWithoutParams := stripParams(commandWithParams); len(commandWithoutParams) >= maxCommandLength {
+			return fmt.Errorf("'%v' command is too long. It shall be less than %v characters", commandWithoutParams, maxCommandLength)
+		}
+
+		if len(commandWithParams) >= maxCommandLengthWithParams {
+			return fmt.Errorf("command with parameters '%v' shall be less than %v characters", commandWithParams, maxCommandLengthWithParams)
+		}
+	}
+
+	return nil
+}
+
+// Gathers all unique commands and processes each command sequentially
+// Parallel processing could be achieved by running several instances of this plugin with different settings
+func (dpdk *dpdk) Gather(accumulator telegraf.Accumulator) error {
+	commands := dpdk.gatherCommands(accumulator)
+
+	for _, command := range commands {
+		dpdk.processCommand(command, accumulator)
+	}
+
+	return nil
+}
+
+// Gathers all unique commands
+func (dpdk *dpdk) gatherCommands(accumulator telegraf.Accumulator) []string {
+	var commands []string
+	if contains(dpdk.DeviceTypes, "ethdev") {
+		commands = append(commands, dpdk.getCommandsAndParamsCombinations(dpdk.ethdevCommands, dpdk.EthdevConfig.EthdevExcludeCommands, ethdevListCommand, accumulator)...)
+	}
+
+	if contains(dpdk.DeviceTypes, "rawdev") {
+		commands = append(commands, dpdk.getCommandsAndParamsCombinations(dpdk.rawdevCommands, []string{}, rawdevListCommand, accumulator)...)
+	}
+
+	commands = append(commands, dpdk.AdditionalCommands...)
+
+	return uniqueValues(commands)
+}
+
+func (dpdk *dpdk) getCommandsAndParamsCombinations(appendedCommands []string, excludedCommands []string, listCommand string, accumulator telegraf.Accumulator) []string {
+	validCommands := removeSubset(appendedCommands, excludedCommands)
+
+	commands, err := dpdk.appendCommandsWithParamsFromList(listCommand, validCommands)
+	if err != nil {
+		accumulator.AddError(fmt.Errorf("error occurred during fetching of %v params - %v", listCommand, err))
+	}
+	return commands
+}
+
+// Fetches all identifiers of devices and then creates all possible combinations of commands for each device
+func (dpdk *dpdk) appendCommandsWithParamsFromList(listCommand string, commands []string) ([]string, error) {
```

Comment:
Please try to follow a consistent parameter ordering. In the functions above the commands come first, but here you apparently switch the order...

Reply:
Right now consistency in parameter ordering is preserved.